cordapp-example/workflows-kotlin/src/test/kotlin/com/example/test/contract/IOUContractTests.kt

```diff
@@ -87,10 +103,86 @@ class IOUContractTests {
         }
     }
-}
\ No newline at end of file
+
+    @Test
+    fun `Destroy - There should be one IOUState input`() {
+        ledgerServices.ledger {
+            transaction {
+                command(lender.publicKey, IOUContract.Commands.Destroy())
+                input(IOUContract.ID, IOUState(iouValue, lender.party, borrower.party))
+                input(IOUContract.ID, IOUState(iouValue, lender.party, borrower.party))
+                `fails with`("There should be one input.")
+            }
+        }
+    }
+
+    @Test
+    fun `Destroy - There should be no IOUState output`() {
+        ledgerServices.ledger {
+            transaction {
+                command(lender.publicKey, IOUContract.Commands.Destroy())
+                input(IOUContract.ID, IOUState(iouValue, lender.party, borrower.party))
+                output(IOUContract.ID, IOUState(iouValue, lender.party, borrower.party))
+                `fails with`("There should be no output.")
+            }
+        }
+    }
+
+    @Test
+    fun `Destroy - Lender must be a signer`() {
+        ledgerServices.ledger {
+            transaction {
+                input(IOUContract.ID, IOUState(iouValue, lender.party, borrower.party))
+
+                tweak {
+                    command(listOf(lender.publicKey, borrower.publicKey), IOUContract.Commands.Destroy())
+                    `fails with`("Expect one signer.")
+                }
+
+                tweak {
+                    command(borrower.publicKey, IOUContract.Commands.Destroy())
+                    `fails with`("Lender must be a signer.")
+                }
+
+                command(lender.publicKey, IOUContract.Commands.Destroy())
+                verifies()
+            }
+        }
+    }
+
+    @Test
+    fun `Destroy accepts other inputs and outputs`() {
+        ledgerServices.ledger {
+            transaction {
+                command(lender.publicKey, IOUContract.Commands.Destroy())
+                input(IOUContract.ID, IOUState(iouValue, lender.party, borrower.party))
+
+                /*
+                    for some reason I can't pass this test
+                    when I use Cash as other input/output I got:
+                    java.lang.IllegalStateException: Required com.example.contract.IOUContract.Commands command
+
+                command(borrower.publicKey, CashContract.Commands.Move())
+                input(CashContract.ID, CashState(iouValue.toLong(), borrower.party))
+                output(CashContract.ID, CashState(iouValue.toLong(), lender.party))
+                // */
+
+                /*
+                    when I'm using DummyContract, it fails because of:
+                    net.corda.core.transactions.MissingContractAttachments Cannot find contract attachments for net.corda.testing.contracts.DummyContract
+
+                command(lender.publicKey,  DummyContract.Commands.Create())
+                input(DummyContract.PROGRAM_ID, DummyState())
+                // */
```

Comment:
about https://github.com/DZariusz/corda-samples/pull/1/files#r399298847
I tried to use Dummy objects to test that I accept other inputs/outputs but no luck... maybe you can give me some tips whats wrong with it?

Reply:
I commented about adding the package to the mock network.